tests/out/msl/resource-binding-map.msl

```diff
@@ -0,0 +1,70 @@
+// language: metal2.0
+#include <metal_stdlib>
+#include <simd/simd.h>
+
+using metal::uint;
+
+struct DefaultConstructible {
+    template<typename T>
+    operator T() && {
+        return T {};
+    }
+};
+
+struct entry_point_oneInput {
+};
+struct entry_point_oneOutput {
+    metal::float4 member [[color(0)]];
+};
+fragment entry_point_oneOutput entry_point_one(
+  metal::float4 pos [[position]]
+, metal::texture2d<float, metal::access::sample> t [[texture(0)]]
+) {
+    constexpr metal::sampler s(
+        metal::s_address::clamp_to_edge,
+        metal::t_address::clamp_to_edge,
+        metal::r_address::clamp_to_edge,
+        metal::mag_filter::linear,
+        metal::min_filter::linear,
+        metal::coord::normalized
+    );
+    metal::float4 _e4 = t.sample(s, pos.xy);
+    return entry_point_oneOutput { _e4 };
+}
+
+
+struct entry_point_twoOutput {
+    metal::float4 member_1 [[color(0)]];
+};
+fragment entry_point_twoOutput entry_point_two(
+  metal::texture2d<float, metal::access::sample> t [[texture(1)]]
+, metal::sampler s [[sampler(1)]]
+, constant metal::float2& uniformOne [[buffer(0)]]
+) {
+    metal::float2 _e3 = uniformOne;
+    metal::float4 _e4 = t.sample(s, _e3);
+    return entry_point_twoOutput { _e4 };
+}
+
+
+struct entry_point_threeOutput {
```

Comment:
`entry_point_three` is a case where `t` and `s` are defined in `per_stage_map` while `uniformOne` and `uniformTwo` are defined `per_entry_point_map`.